Reinterop~/MethodsImplementedInCpp.cs

```diff
@@ -19,7 +19,7 @@ public static void Generate(CppGenerationContext context, TypeToGenerate item, G
             CppType objectHandleType = CppObjectHandle.GetCppType(context);
 
             // We only need a C++ instance if any partial methods are non-static
-            bool needsInstance = item.MethodsImplementedInCpp.Any(m => !m.IsStatic);
+            bool needsInstance = !item.ImplementationStaticOnly && item.MethodsImplementedInCpp.Any(m => !m.IsStatic);
```

Comment:
Is it possible to for someone to (incorrectly) declare a static-only implementation and declare a non-static partial method? Is there a check for that ?

Reply:
That's actually allowed! It just means the the _static_ C++ implementation will receive the C# _instance_ as its first parameter.

Reply:
ah, I think I understand! easy to lose track of what's static and what's an instance 😅